### PR TITLE
Upgrade homebrew formula to 1.0-rc.2

### PR DESCRIPTION
**Pull request by mberhault**
For [1.0-rc.2 release](https://github.com/cockroachdb/cockroach/issues/15660)

### CODE REVIEW

**Comment by benesch** [comment]
That reminds me! The source tarball's Makefile has gotten even smarter, so we need to also update the `install` method to look like this:

```
def install
  system "make", "install", "prefix=" + prefix
end
```

**Comment by mberhault** [comment]
I'm trying to test it following the steps in https://github.com/cockroachdb/cockroach/wiki/Release-Process, but `brew install cockroach` keeps resetting to `master` so it pulling in rc1. thoughts?

**Comment by benesch** [comment]
Weird! Sorry, on a train. Lemme check out your branch and see if it works.

File: Formula/cockroach.rb [hunk 1]
```diff
@@ -1,19 +1,17 @@
...
 
   depends_on "cmake" => :build
   depends_on "go" => :build
   depends_on "xz" => :build
 
   def install
-    ENV["GOPATH"] = buildpath
-    system "make", "install"
-    bin.install "bin/cockroach" => "cockroach"
+    system "make", "install", "prefix=" + prefix
```
Comment (tamird) [inline Formula/cockroach.rb]: if y'all are making more changes, care to appease me with "prefix=${prefix}"? thanks!

Comment (benesch) [inline Formula/cockroach.rb]: Eh? It's Ruby! Do you mean "prefix=#{prefix}"?

Comment (tamird) [inline Formula/cockroach.rb]: Erhm, yes!

**Comment by benesch** [comment]
@marc, caught a bug in the release process instructions. Should have said `cd $(brew --repo cockroachdb/cockroach)`, not `cd $(brew --repo cockroach)`. That should have failed loudly though. Not sure why yours insisted on picking up rc.1—the rest of the instructions work as-is for me.

Tarball download's at 80%. I'll report back shortly.

**Comment by mberhault** [comment]
oh, right, I corrected that myself. However, it `brew install` insists on switching me back to `master`:
```
(marc/upgrade_brew)$ brew install cockroach
==> Installing cockroach from cockroachdb/cockroach
==> Downloading https://binaries.cockroachdb.com/cockroach-v1.0-rc.1.src.tgz
Already downloaded: /Users/marc/Library/Caches/Homebrew/cockroach-1.0-rc.1.tgz
^C
(master)$
```

**Comment by benesch** [comment]
That's so odd; it just doesn't do that for me. You might also try `brew upgrade cockroach`. Or perhaps this whole scheme only works if you already have an old version of the formula installed.

Also, bad news: I'm afraid I've flubbed. As-written, Make will complain that it doesn't recognize the `prefix` variable. Here's a patch to include a patch (heh) to fix it:

```diff
diff --git a/Formula/cockroach.rb b/Formula/cockroach.rb
index 0acf248d..738a5c59 100644
--- a/Formula/cockroach.rb
+++ b/Formula/cockroach.rb
@@ -10,6 +10,8 @@ class Cockroach < Formula
   depends_on "go" => :build
   depends_on "xz" => :build
 
+  patch :DATA
+
   def install
     system "make", "install", "prefix=#{prefix}"
   end
@@ -78,3 +80,33 @@ class Cockroach < Formula
     end
   end
 end
+__END__
+--- a/src/github.com/cockroachdb/cockroach/build/common.mk
++++ b/src/github.com/cockroachdb/cockroach/build/common.mk
+@@ -181,7 +181,7 @@ endif
+ #     TARGET-NAME: [export] VARIABLE [:+?]=
+ #
+ # The additional complexity below handles whitespace and comments.
+-$(REPO_ROOT)/build/variables.mk: $(REPO_ROOT)/Makefile $(REPO_ROOT)/.go-version $(REPO_ROOT)/build/common.mk
++$(REPO_ROOT)/build/variables.mk: $(REPO_ROOT)/Makefile $(REPO_ROOT)/.go-version $(REPO_ROOT)/build/common.mk $(REPO_ROOT)/build/archive/contents/Makefile
+ 	@echo '# This file is auto-generated by Make.' > $@
+ 	@echo '# DO NOT EDIT!' >> $@
+ 	@echo 'define VALID_VARS' >> $@
+--- a/src/github.com/cockroachdb/cockroach/build/variables.mk
++++ b/src/github.com/cockroachdb/cockroach/build/variables.mk
+@@ -37,6 +37,7 @@ define VALID_VARS
+   GOVERS
+   GO_INSTALL
+   HOST_TRIPLE
++  INSTALL
+   ISDARWIN
+   JEMALLOC_DIR
+   JEMALLOC_SRC_DIR
+@@ -80,5 +81,7 @@ define VALID_VARS
+   XHOST_BIN_DIR
+   XHOST_TRIPLE
+   YARN_INSTALLED_TARGET
++  bindir
++  prefix
+   space
+ endef
```

I guess we should cherry pick that for 1.0? Agh. 🤦‍♂️

**Comment by benesch** [comment]
With the patch, though, all is well:

```
(marc/upgrade_brew * u=)$ ls -l $(which cockroach)
lrwxr-xr-x 1 benesch admin 42 May  5 11:37 /usr/local/bin/cockroach -> ../Cellar/cockroach/1.0-rc.2/bin/cockroach

(marc/upgrade_brew * u=)$ cockroach version
Build Tag:    v1.0-rc.2
Build Time:   2017/05/05 15:37:01
Distribution: CCL
Platform:     darwin amd64
Go Version:   go1.8.1-parallelbuilds
C Compiler:   4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)
Build SHA-1:  1700211837c3e519bf093f8cda26fe2afb028ff3
Build Type:   development
```

**Comment by mberhault** [comment]
does it still work with the previous install step? if so, this can wait.

**Comment by benesch** [comment]
Those could be convinced to work with some adjustments.

**Comment by benesch** [comment]
Perhaps:

```bash
def install
  system "make", "build"
  bin.install "src/github.com/cockroachdb/cockroach/cockroach" => "cockroach"
end
```

Let me test.

**Comment by benesch** [comment]
Confirmed to work with this smaller patch.

```diff
diff --git a/Formula/cockroach.rb b/Formula/cockroach.rb
index 0acf248d..4885927a 100644
--- a/Formula/cockroach.rb
+++ b/Formula/cockroach.rb
@@ -11,7 +11,8 @@ class Cockroach < Formula
   depends_on "xz" => :build
 
   def install
-    system "make", "install", "prefix=#{prefix}"
+    system "make", "build"
+    bin.install "src/github.com/cockroachdb/cockroach/cockroach" => "cockroach"
   end
 
   def caveats; <<-EOS.undent
```

**Comment by mberhault** [comment]
ok. let's do that instead. We can patch 1.0, but I prefer to avoid patches inside the formula. I'll update it.

**Comment by benesch** [comment]
👍 sorry for the schlep

**Comment by mberhault** [comment]
no worries. PTAL.

**Comment by benesch** [comment]
Looks spot on, but let me do try building just to be absolutely sure.

**Comment by benesch** [comment]
All good. Merge away!

**Pull request merged**